tests/ui/UnreadIndicatorsTest.js

```diff
@@ -0,0 +1,521 @@
+
+                // Tap on the chat again
+                return navigateToSidebarOption(renderedApp, 0);
+            })
+            .then(() => {
+                // Verify the unread indicator is not present
+                const unreadIndicator = renderedApp.queryAllByA11yLabel('New message line indicator');
+                expect(unreadIndicator.length).toBe(0);
+                expect(isDrawerOpen(renderedApp)).toBe(false);
+
+                // Scroll and verify that the new messages badge is also hidden
+                scrollUpToRevealNewMessagesBadge(renderedApp);
+                expect(isNewMessagesBadgeVisible(renderedApp)).toBe(false);
+            });
+    });
+
+    it('Shows a browser notification and bold text when a new message arrives for a chat that is read', () => {
+        let renderedApp;
+        return signInAndGetAppWithUnreadChat()
+            .then((testInstance) => {
+                renderedApp = testInstance;
+
+                // Simulate a new report arriving via Pusher along with reportActions and personalDetails for the other participant
+                const NEW_REPORT_ID = 2;
+                const NEW_REPORT_CREATED_MOMENT = moment();
+                Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${NEW_REPORT_ID}`, {
+                    reportID: NEW_REPORT_ID,
+                    reportName: 'Chat Report',
+                    maxSequenceNumber: 1,
+                    lastReadSequenceNumber: 0,
+                    lastMessageTimestamp: NEW_REPORT_CREATED_MOMENT.utc(),
+                    lastMessageText: 'Comment 1',
+                    participants: [USER_C_EMAIL],
+                });
+                Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${NEW_REPORT_ID}`, {
+                    0: {
+                        actionName: CONST.REPORT.ACTIONS.TYPE.CREATED,
+                        automatic: false,
+                        sequenceNumber: 0,
+                        timestamp: NEW_REPORT_CREATED_MOMENT.unix(),
+                        reportActionID: NumberUtils.rand64(),
+                    },
+                    1: {
+                        actionName: CONST.REPORT.ACTIONS.TYPE.ADDCOMMENT,
+                        actorEmail: USER_C_EMAIL,
+                        person: [{type: 'TEXT', style: 'strong', text: 'User C'}],
+                        sequenceNumber: 1,
+                        timestamp: NEW_REPORT_CREATED_MOMENT.add(5, 'seconds').unix(),
+                        message: [{type: 'COMMENT', html: 'Comment 1', text: 'Comment 1'}],
+                        reportActionID: NumberUtils.rand64(),
+                    },
+                });
+                Onyx.merge(ONYXKEYS.PERSONAL_DETAILS, {
+                    [USER_C_EMAIL]: TestHelper.buildPersonalDetails(USER_C_EMAIL, USER_C_ACCOUNT_ID, 'C'),
+                });
+                return waitForPromisesToResolve();
```

Comment:
Something else I did in the LHN tests was to use `return Onyx.multiSet()` which would prevent you from needing to deal with three separate promises here and can remove another `waitForPromisesToResolve`.

But in that case, I think we also need to wrap them in `act()` to ensure all the component tree is updated properly before making assertions. I have yet to test that out fully.

That also assume that using `set()` is OK (which I think it usually is for tests).

Reply:
Gonna leave this one - mainly because having a few different merges happen seems closer to what would happen in the UI.